### PR TITLE
Added request to update the users awards

### PR DESCRIPTION
**Pull request by lrrrs**

### CODE REVIEW

File: XNGAPIClient/XNGAPIObjectAward.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#import <Foundation/Foundation.h>
+
+@interface XNGAPIObjectAward : NSObject
+@property(nonatomic) NSString *name;
```
Comment (tammofreese) [inline XNGAPIClient/XNGAPIObjectAward.h]: `copy` is your friend.

Comment (tammofreese) [inline XNGAPIClient/XNGAPIObjectAward.h]: Please add a space before the `(`

File: XNGAPIClient/XNGAPIObjectAward.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+@property(nonatomic) NSString *year;
+@property(nonatomic) NSString *url;
+
+-(instancetype)initWithName:(NSString*)awardName dateAwarded:(NSString*)awardYear url:(NSString*)awardURL;
```
Comment (tammofreese) [inline XNGAPIClient/XNGAPIObjectAward.h]: The rest of the API uses a space before the pointer symbol, please follow that here as well. I wonder why uncrustify hasn't caught this.

File: XNGAPIClient/XNGAPIObjectAward.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#import <Foundation/Foundation.h>
+
+@interface XNGAPIObjectAward : NSObject
+@property(nonatomic) NSString *name;
+@property(nonatomic) NSString *year;
+@property(nonatomic) NSString *url;
```
Comment (tammofreese) [inline XNGAPIClient/XNGAPIObjectAward.h]: How are URLs handled in the rest of the API to outside? I am fine with a string here if it is for consistency.

File: XNGAPIClient/XNGAPIClient+ProfileEditing.m [hunk 4]
```diff
@@ -437,6 +437,25 @@ - (void)putUpdatePrimaryCompanyWithID:(NSString *)companyID
     [self putJSONPath:path JSONParameters:parameters success:success failure:failure];
 }
 
+- (void)putUpdateAwards:(NSArray<XNGAPIObjectAward*> *)awards
+                success:(void (^)(id JSON))success
+                failure:(void (^)(NSError *error))failure {
+    if(!awards || awards.count == 0) {
```
Comment (tammofreese) [inline XNGAPIClient/XNGAPIClient+ProfileEditing.m]: `awards.count == 0` is enough, this is not Swift yet ;)
Also, please add a space after `if` to align with the rest of the API code.

**Comment by tammofreese** [comment]
:+1: from my side.

File: Example/ExampleTests/XNGProfileEditingTests.m [hunk 5]
```diff
@@ -24,6 +25,34 @@ - (void)tearDown {
     [self.testHelper tearDown];
 }
 
+-(void)testUpdateAwards {
+    
+    NSArray *awards = @[[[XNGAPIAward alloc] initWithName:@"Award" dateAwarded:@"2016" url:[NSURL URLWithString:@"www.award.de"]],
+                        [[XNGAPIAward alloc] initWithName:@"Bward" dateAwarded:@"2015" url:[NSURL URLWithString:@"www.bward.de"]]];
```
Comment (m0rk) [inline Example/ExampleTests/XNGProfileEditingTests.m]: It is maybe nicer to format it like
```objc
NSArray *awards = @[
    [[XNGAPIAward alloc] initWithName:@"Award" dateAwarded:@"2016" url:[NSURL URLWithString:@"www.award.de"]],
    [[XNGAPIAward alloc] initWithName:@"Bward" dateAwarded:@"2015" url:[NSURL URLWithString:@"www.bward.de"]]
];
```

File: Example/ExampleTests/XNGProfileEditingTests.m [hunk 6]
```diff
@@ -3,6 +3,7 @@
 #import <XNGAPIClient/UIImage+Base64Encoding.h>
 #import <XNGAPIClient/XNGAPI.h>
 #import <XNGAPIClient/XNGAPIClient+Misc.h>
+#import <XNGAPIClient/XNGAPIClient+ProfileEditing.h>
```
Comment (m0rk) [inline Example/ExampleTests/XNGProfileEditingTests.m]: Maybe put `#import <XNGAPIClient/XNGAPIClient+Misc.h>` and `#import <XNGAPIClient/XNGAPIClient+ProfileEditing.h>` in to the umbrella header `<XNGAPIClient/XNGAPI.h>`?

File: Example/ExampleTests/XNGProfileEditingTests.m [hunk 7]
```diff
@@ -24,6 +25,34 @@ - (void)tearDown {
...
+    [self.testHelper executeCall:^{
+        [[XNGAPIClient sharedClient] putUpdateAwards:awards success:nil failure:nil];
+    } withExpectations:^(NSURLRequest *request, NSMutableDictionary *query, NSMutableDictionary *body) {
+        expect(request.URL.host).to.equal(@"api.xing.com");
+        expect(request.URL.path).to.equal(@"/v1/users/me/professional_experience/awards");
+        expect(request.HTTPMethod).to.equal(@"PUT");
+        expect([query allKeys]).to.haveCountOf(0);
+        expect([body valueForKey:@"awards"]).to.haveCountOf(2);
+        
+        NSArray<NSDictionary*>* awards = [body valueForKey:@"awards"];
```
Comment (m0rk) [inline Example/ExampleTests/XNGProfileEditingTests.m]: Just a whitespave issue. 😏 
```objc
NSArray<NSDictionary *>* awards = [body valueForKey:@"awards"];
```

File: XNGAPIClient/XNGAPIClient+ProfileEditing.m [hunk 8]
```diff
@@ -437,6 +437,25 @@ - (void)putUpdatePrimaryCompanyWithID:(NSString *)companyID
     [self putJSONPath:path JSONParameters:parameters success:success failure:failure];
 }
 
+- (void)putUpdateAwards:(NSArray<XNGAPIAward *> *)awards
+                success:(void (^)(id JSON))success
+                failure:(void (^)(NSError *error))failure {
+    if(awards.count == 0) {
```
Comment (m0rk) [inline XNGAPIClient/XNGAPIClient+ProfileEditing.m]: Whitespace missing
```objc
if (awards.count == 0) {
```

**Comment by m0rk** [comment]
:shipit:!

**Pull request merged**